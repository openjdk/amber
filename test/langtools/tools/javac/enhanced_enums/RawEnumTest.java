/*
 * @test /nodynamioccopyright/
 * @summary Smoke test for use of raw generic enums
 * @compile/fail/ref=RawEnumTest.out -Xlint:all -XDrawDiagnostics RawEnumTest.java
 */

import java.util.*;
import java.util.stream.Stream;

class RawEnumTest {
    interface A<X> {
        void m(X x);
    }

    enum Foo<X> implements A<String> {
        A<String>, B<Integer>;

        String m() { return ""; }
        public void m(String s) { }
        public List<X> list(List<X> l) { return null; }
        public List<String> list2(List<String> l) { return null; }
    }

    void test() {
        EnumSet<Foo> fooSet = EnumSet.noneOf(Foo.class); //ok
        Foo f = null; //no raw type warning
        f.m(""); //ok
        f.m(1); //error
        f.list(null); //unchecked
        f.list2(null); //ok
        Foo<?> foow = f; //ok
        Foo<String> fs = f; //unchecked
        fooSet = EnumSet.of(Foo.A, Foo.B); //ok
        Stream<Foo> fooStream = Arrays.stream(Foo.values());
    }
}
