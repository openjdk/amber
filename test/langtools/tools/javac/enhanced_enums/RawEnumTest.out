RawEnumTest.java:28:10: compiler.err.cant.apply.symbols: kindname.method, m, int,{(compiler.misc.inapplicable.method: kindname.method, RawEnumTest.Foo, m(), (compiler.misc.arg.length.mismatch)),(compiler.misc.inapplicable.method: kindname.method, RawEnumTest.Foo, m(java.lang.String), (compiler.misc.no.conforming.assignment.exists: (compiler.misc.inconvertible.types: int, java.lang.String)))}
RawEnumTest.java:29:15: compiler.warn.unchecked.call.mbr.of.raw.type: list(java.util.List<X>), RawEnumTest.Foo
RawEnumTest.java:32:26: compiler.warn.prob.found.req: (compiler.misc.unchecked.assign), RawEnumTest.Foo, RawEnumTest.Foo<java.lang.String>
1 error
2 warnings
